### DISCUSSION TITLE
Models suitable for changing context

### DISCUSSION THREAD

**Discussion by Joe-W-Davies** [recommenders-team/recommenders#2152] [Q&A]
Hi all,

For the neural network based models in this repository, it seems like adding context is only possible when the contextual feature is static. For example in the MovieLens dataset, the [Wide and Deep example notebook ](https://github.com/recommenders-team/recommenders/blob/main/examples/00_quick_start/wide_deep_movielens.ipynb) only considers "genre" of the item as an additional context feature. This item feature is the same for every user-item pair, given that the movie genre doesn't change.

I'm interested in the case where we have dynamic context features that change for each user-item interaction. For example, the age of the movie in days would be a dynamic feature, because it will be likely different for each user-item pair. The day of the week the interaction occurs on is another example.

I'm aware Boosted Tree models can incorporate these kind of features, but in my experience they are poorer at handling the user and item IDs; hence why I want to use a NN-based approach, with better embeddings for IDs.

Are there any models suitable for this application in the repo? 

Thank you!

**Comment by daviddavo** [comment]
I understand that factorization machines should be able to do the job, but I still haven't used them

[Quickstart xDeepFM notebook](https://github.com/recommenders-team/recommenders/blob/main/examples/00_quick_start/xdeepfm_criteo.ipynb)
